arroyo-api/src/connection_tables.rs

```diff
@@ -133,81 +335,118 @@ pub(crate) async fn create(
+            Some(schema) => schema,
+            None => {
+                warn!("No schema found for table {}", self.id);
+                return Err(());
+            }
+        };
 
-    Ok(rx)
+        Ok(ConnectionTable {
+            id: self.id,
+            pub_id: self.pub_id,
+            name: self.name,
+            connection_profile: None,
+            connector: self.connector,
+            table_type: self.table_type.into(),
+            config: serde_json::to_string(&self.config).unwrap(),
+            schema,
+            consumers: self.consumer_count as u32,
+        })
+    }
 }
 
-fn get_connection(c: &GetConnectionTables, connector: &dyn ErasedConnector) -> Option<Connection> {
-    let config = serde_json::to_string(&c.connection_config.as_ref()?).unwrap();
-    Some(Connection {
-        id: format!("{}", c.connection_id?),
-        name: c.connection_name.as_ref()?.clone(),
-        connector: c.connection_type.as_ref()?.clone(),
-        description: connector.config_description(&config).ok()?,
-        config,
-    })
-}
+/// List all connection tables
+#[utoipa::path(
+    get,
+    path = "/v1/connection_tables",
+    tag = "connection_tables",
+    responses(
+        (status = 200, description = "Got connection table collection", body = ConnectionTableCollection),
+    ),
+)]
+pub(crate) async fn get_connection_tables(
+    State(state): State<AppState>,
+    bearer_auth: BearerAuth,
+) -> Result<Json<ConnectionTableCollection>, ErrorResp> {
+    let client = client(&state.pool).await?;
+    let auth_data = authenticate(&state.pool, bearer_auth).await?;
 
-pub(crate) async fn get<C: GenericClient>(
-    auth: &AuthData,
-    client: &C,
-) -> Result<Vec<ConnectionTable>, Status> {
     let tables = api_queries::get_connection_tables()
-        .bind(client, &auth.organization_id)
+        .bind(&client, &auth_data.organization_id)
         .all()
         .await
         .map_err(log_and_map)?;
 
-    Ok(tables
+    let data: Vec<ConnectionTable> = tables
         .into_iter()
         .filter_map(|t| {
             let Some(connector) = connector_for_type(&t.connector) else {
                 warn!("invalid connector {} in saved ConnectionTable {}", t.connector, t.id);
                 return None;
             };
 
-            let connection = get_connection(&t, &*connector);
+            let connection = get_connection_profile(&t, &*connector);
 
             let table = serde_json::to_string(&t.config).unwrap();
-            let schema = t.schema.map(|s| serde_json::from_value(s).unwrap());
+
+            let schema: Option<ConnectionSchema> =
+                t.schema
+                    .and_then(|value| match serde_json::from_value(value) {
+                        Ok(connection_schema) => Some(connection_schema),
+                        Err(err) => {
+                            warn!("Connection schema deserialization failed: {:?}", err);
+                            None
```

Comment:
Similarly, we should not return partial connection tables — if we fail to deserialize the schema we should warn and then omit the table

Reply:
I agree, done.